File: generated-code/example/go/model/enumeration.go
package model

import "io"
import . "trans_gen_test/stream"

type Enumeration int32
const (
    EnumerationValueOne Enumeration = 0
    EnumerationValueTwo Enumeration = 1
)
func ReadEnumeration(reader io.Reader) Enumeration {
    switch ReadInt32(reader) {
    case 0:
        return EnumerationValueOne
    case 1:
        return EnumerationValueTwo
    }
    panic("Unexpected tag value")
}

File: generated-code/example/rust/src/model/enumeration.rs
use super::*;

#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub enum Enumeration {
    ValueOne,
    ValueTwo,
}

impl trans::Trans for Enumeration {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        let tag: i32 = match self {
            Self::ValueOne => 0,
            Self::ValueTwo => 1,
        };
        trans::Trans::write_to(&tag, writer)
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let tag = <i32 as trans::Trans>::read_from(reader)?;
        match tag {
            0 => Ok(Self::ValueOne),
            1 => Ok(Self::ValueTwo),
            _ => Err(std::io::Error::new(
                std::io::ErrorKind::Other,
                format!("Unexpected tag {:?}", tag))),
        }
    }
}
